lib/daru/dataframe.rb

```diff
@@ -1835,14 +1839,14 @@ def prepare_vector_for_insert name, vector
     end
 
     def insert_or_modify_row indexes, vector
-      vector = 
-        unless vector.is_a? Daru::Vector
+      vector =
+        if !vector.is_a? Daru::Vector
           Daru::Vector.new vector
         else
           vector.reindex @vectors
         end
```

Comment:
Exchange places of if and else branches :) This way you'll not need "if not..." (which is hard to read with "...else", and THAT is real sense of complaining about `unless`/`else` combination, not just replacing one operator with another).